core/src/apps/common/passphrase.py

```diff
@@ -17,6 +17,9 @@ def is_enabled() -> bool:
 
 async def get_passphrase(msg: ThpCreateNewSession) -> str:
     if not is_enabled():
+        if msg.passphrase:
+            # passphrase is not None/empty
+            raise DataError("Passphrase protection is disabled")
         return ""
 
     if msg.on_device or storage_device.get_passphrase_always_on_device():
```

Comment:
while we're here:
this should be `and msg.passphrase is None`

i recall specifying the logic of the function somewhere:
1. `if passphrase is not None and not on_device`: use contents of `passphrase` field
2. `if passphrase is None and on_device`: ask the user
3. `if passphrase is None and not on_device`: create a seedless session
4. `if passphrase is not None and on_device`: raise an error